Fixes Dociler and bullpup grenade launcher

emmanuelbassil opened the pull request:
🆑 emmanuelbassil
bugfix: Docilers now properly self-recharge
bugfix: Can now properly disable safety on underslung grenade launcher.
bugfix: Light bullpup can no longer be loaded with grenades it can't even fire. It also no longer insists it has a grenade launcher despite best attempts to convince it otherwise.
/🆑 

In this PR, I also made attack() under dociler use booleans and ITEM_TRY_TO_ATTACK in anticipation of my upcoming PR with 60-65 attack() changes. One less to review! This one doesn't call parent attack yet, but it will in the other PR once the attack() and use_weapon() stuff is properly sorted out.

Also made self-charging energy guns stop processing all the time.

emmanuelbassil commented:
Fixed unit tests; ready for review.

MuckerMayhem commented on code/game/objects/items/devices/dociler.dm:
```diff
@@ -12,8 +13,22 @@
+	loaded = 1
+	icon_state = "animal_tagger1"
+	update_icon()
+	src.visible_message("\The [src] beeps, refilling itself.")
```
```suggestion
	visible_message("\The [src] beeps, refilling itself.")
```

MuckerMayhem commented on code/game/objects/items/devices/dociler.dm:
```diff
@@ -12,8 +13,22 @@
+/obj/item/device/dociler/Process()
+	if (loaded)
+		return PROCESS_KILL
+	if(++charge_tick < recharge_time)
```
```suggestion
	if (++charge_tick < recharge_time)
```

MuckerMayhem commented on code/game/objects/items/devices/dociler.dm:
```diff
@@ -27,34 +42,35 @@
-		to_chat(user, SPAN_WARNING("\The [src] isn't loaded!"))
-		return
+		user.visible_message("\The [user] thrusts \the [src] deep into \the [L]'s head, injecting something!")
+		to_chat(L, SPAN_NOTICE("You feel pain as \the [user] injects something into you. All of a sudden you feel as if [user] is the friendliest and nicest person you've ever know. You want to be friends with him and all his friends."))
```
```suggestion
		to_chat(L, SPAN_NOTICE("You feel pain as \the [user] injects something into you. All of a sudden you feel as if [user] is the friendliest and nicest person you've ever known. You want to be friends with them and all their friends."))
```

juxjux9930 commented on code/game/objects/items/devices/dociler.dm:
```diff
@@ -27,34 +42,35 @@
+		if (mode == "somewhat")
+			L.faction = user.faction
+		else
+			L.faction = null
```
nulling the faction might make several animals hostile instead of docile

emmanuelbassil replied:
Will just make it change to user faction for now. This mode stuff is superfluous.

emmanuelbassil committed:
Fix Dociler self-recharging


Fixes


Auto stash before rebase of "refs/heads/dociler"
Auto stash before rebase of "refs/heads/eyestab"
Auto stash before rebase of "refs/heads/dociler"
Fixes

emmanuelbassil commented:
Fixed!
